src/story_protocol_python_sdk/resources/IPAccount.py

```diff
@@ -153,4 +156,85 @@ def owner(self, ip_id: str) -> str:
+                raise ValueError(f"IP id {ip_id} is not registered")
+
+            data = self.core_metadata_module_client.contract.encode_abi(
+                abi_element_identifier="setMetadataURI",
+                args=[Web3.to_checksum_address(ip_id), metadata_uri, metadata_hash]
+            )
+
+            response = self.execute(
+                to=self.core_metadata_module_client.contract.address,
+                value=0,
+                ip_id=ip_id,
+                data=data,
+                tx_options=tx_options
+            )
+
+            return response
+        except Exception as e:
+            raise e
+
+    def transferERC20(self, ip_id: str, tokens: list, tx_options: dict = None) -> dict:
+        """Transfers ERC20 tokens from the IP Account to the target address.
+
+        :param ip_id str: The IP ID to transfer tokens from.
+        :param tokens list: A list of dictionaries containing token transfer details.
+                           Each dictionary should have 'address' (token contract address),
+                           'target' (recipient address), and 'amount' (token amount).
+        :param tx_options dict: [Optional] The transaction options.
+        :returns dict: A dictionary with the transaction hash.
+        :raises ValueError: If the IP ID is invalid or not registered, or if token parameters are invalid.
+        """
+        try:
+            if not self._is_registered(ip_id):
+                raise ValueError(f"IP id {ip_id} is not registered")
+
+            ip_account = IPAccountImplClient(self.web3, contract_address=ip_id)
+
+            # Since we don't have executeBatch, we'll execute transfers one by one
+            results = []
+            for token in tokens:
+                if not all(key in token for key in ['address', 'target', 'amount']):
+                    raise ValueError("Each token transfer must include 'address', 'target', and 'amount'")
+                
+                token_address = self.web3.to_checksum_address(token['address'])
+                target_address = self.web3.to_checksum_address(token['target'])
+                amount = int(token['amount'])
+                
+                data = self.mock_erc20_client.contract.encode_abi(
+                    abi_element_identifier="transfer", 
+                    args=[target_address, amount]
+                )
+                
+                response = build_and_send_transaction(
+                    self.web3,
+                    self.account,
+                    ip_account.build_execute_transaction,
+                    self.web3.to_checksum_address(token_address),
+                    0,
+                    data,
+                    0
+                )
+                results.append(response)
+            
+            # Return the hash of the last transaction
+            # In a real implementation with executeBatch, we would return a single transaction hash
```

Comment:
why not just use `executeBatch`? it should be pretty straightwards to support right? a single tx still have benefits in terms of speed and cost so if easy to support we should support it.